analyzers/src/SonarAnalyzer.CSharp/Rules/FrameworkViewCompiler.cs

```diff
@@ -0,0 +1,339 @@
+        var razorCompiler = new RazorCompiler(ROOT_DIR, filesProvider);
+        var dummyCompilation = compilation;
+
+        foreach (var razorDocument in razorCompiler.CompileAll().Where(x => x.Source.FilePath.Contains("Contact")))
+        {
+            if (razorDocument.GetCSharpDocument()?.GeneratedCode is { } csharpCode)
+            {
+                var fp = dummyCompilation.SyntaxTrees.FirstOrDefault(x => x.FilePath.Contains("Contact")).FilePath;
+
+                var razorTree = CSharpSyntaxTree.ParseText(csharpCode, new CSharpParseOptions(compilation.GetLanguageVersion()))
+                    .WithFilePath(fp);
+                dummyCompilation = dummyCompilation.AddSyntaxTrees(razorTree);
+            }
+        }
+
+        return dummyCompilation;
+    }
+}
+#region RULE_FINDER
+// Copied from the test framework.
+internal static class RuleFinder2
+{
+    public static IEnumerable<Type> AllAnalyzerTypes { get; }       // Rules and Utility analyzers
+    public static IEnumerable<Type> AllTypesWithDiagnosticAnalyzerAttribute { get; }
+    public static IEnumerable<Type> RuleAnalyzerTypes { get; }      // Rules-only, without Utility analyzers
+    public static IEnumerable<Type> UtilityAnalyzerTypes { get; }
+    public static IEnumerable<Type> CodeFixTypes { get; }
+
+    static RuleFinder2()
+    {
+        var allTypes = new[]
+        {
+            typeof(FlagsEnumZeroMember),
+            typeof(FlagsEnumZeroMemberBase<int>)
+        }
+            .SelectMany(x => x.Assembly.GetExportedTypes())
+            .ToArray();
+        CodeFixTypes = allTypes.Where(x => typeof(CodeFixProvider).IsAssignableFrom(x) && x.GetCustomAttributes<ExportCodeFixProviderAttribute>().Any()).ToArray();
+        AllAnalyzerTypes = allTypes.Where(x => x.IsSubclassOf(typeof(DiagnosticAnalyzer)) && x.GetCustomAttributes<DiagnosticAnalyzerAttribute>().Any()).ToArray();
+        AllTypesWithDiagnosticAnalyzerAttribute = allTypes.Where(x => x.GetCustomAttributes<DiagnosticAnalyzerAttribute>().Any()).ToArray();
+        UtilityAnalyzerTypes = AllAnalyzerTypes.Where(x => typeof(UtilityAnalyzerBase).IsAssignableFrom(x)).ToList();
+        RuleAnalyzerTypes = AllAnalyzerTypes.Except(UtilityAnalyzerTypes).ToList();
+    }
+
+    public static IEnumerable<Type> GetAnalyzerTypes(AnalyzerLanguage language) =>
+        RuleAnalyzerTypes.Where(x => TargetLanguage(x) == language);
+
+    public static IEnumerable<DiagnosticAnalyzer> CreateAnalyzers(AnalyzerLanguage language, bool includeUtilityAnalyzers)
+    {
+        var types = GetAnalyzerTypes(language);
+        if (includeUtilityAnalyzers)
+        {
+            types = types.Concat(UtilityAnalyzerTypes.Where(x => TargetLanguage(x) == language));
+        }
+        foreach (var type in types.Where(x => x == typeof(VariableUnused))) // TODO: Remove ".Where(x => x == typeof(VariableUnused))" and see error. Locally it breaks for me with StackOverflow, at a compiled Regex.
```

Comment:
Check TODO.

We will probably need to exclude some of our rules.
For now, we can add one or two just to test. I chose S1481 because it's really braindead and easy to add issues for.